packages/ai/src/embed/embed.ts

```diff
@@ -168,6 +173,8 @@ Only applicable for HTTP-based providers.
         value,
         embedding,
         usage,
+        providerMetadata: (response?.body as EmbeddingResponseBody)
```

Comment:
Same comment as above re: avoid use of `as` -- explore type safety following pattern in existing similar code/functions.